pkg/domain/network_change_event.go

```diff
@@ -0,0 +1,55 @@
+package domain
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+// NetworkChangeEvent represent a result row of database query
+type NetworkChangeEvent struct {
+	ResourceID string
+	IPAddress  string
+	Hostname   sql.NullString
```

Comment:
Echoing @zlozano , bundling the implementation details with the domain type leaks the abstraction more than we typically allow. Our usual pattern for this is to stick to standard lib types (as much as possible) for domain types and have the implementation specific details hidden by using unexported types in the implementation packages. For example:

```golang
// domain/something.go
type Change struct {
 Hostname *string
}
```

```golang
// someting/pgsql.go
type pgsqlChange struct{
 Hostname sql.NullString
}
func (s *PgsqlSomething) StoreChange(ctx context.Context, ch Change) error {
  pgch := pgsqlChange{
    Change: sql.NullString(ch.Hostname),
  },
  // ...
}
```

It adds a small layer of tedium to adapt the types but, largely, helps us prevent making implementation specific domain types that have to change every time we change our underlying data model.

Reply:
Oh I didn't realize I left that in there.  I'll take it out.